web-common-marlowe/src/Marlowe/Slot.purs

```diff
@@ -35,21 +42,23 @@ shelleyLaunchDate =
   in
     unsafePartial $ fromJust $ toDateTime <$> instant epoch
 
+secondsDiff :: Slot -> Slot -> Seconds
+secondsDiff a b = Seconds $ BigInteger.toNumber $ unwrap $ a - b
+
 slotToDateTime :: Slot -> Maybe DateTime
 slotToDateTime slot =
   let
-    secondsDiff :: Seconds
-    secondsDiff = Seconds $ BigInteger.toNumber $ unwrap $ slot - shelleyInitialSlot
+    secondsDiff' = secondsDiff slot shelleyInitialSlot
   in
-    adjust secondsDiff shelleyLaunchDate
+    adjust secondsDiff' shelleyLaunchDate
 
 dateTimeToSlot :: DateTime -> Slot
 dateTimeToSlot datetime =
   let
-    secondsDiff :: Seconds
-    secondsDiff = diff datetime shelleyLaunchDate
+    secondsDiff' :: Seconds
+    secondsDiff' = diff datetime shelleyLaunchDate
   in
-    shelleyInitialSlot + (Slot $ BigInteger.fromInt $ round $ unwrap secondsDiff)
+    shelleyInitialSlot + (Slot $ BigInteger.fromInt $ round $ unwrap secondsDiff')
```

Comment:
It doesn't look like this `Seconds` constructor is giving us much since everything has to be done from scratch anyway and it does lose the precision of `BigInteger`

Reply:
Mhmhmh, not sure if we can avoid it. The `DateTime -> Slot` and `Slot -> Maybe DateTime` are used to convert from absolute times, for example the date pickers when we create a new contract give us an absolute time that we need to convert to a Slot. And when we display timeouts, we convert from Slot to DateTime and use the datetime formatters to get the string.

The only place where we could get rid of is in humanizeDuration, where it ask for the difference of two slot's (which are already in seconds) and then do some manual calculations to see how to represent the difference as `1hr 20min`.

Reply:
Oh, you are right, here you do use `diff` too. Yes, I guess it is only `humanizeDuration`. I was looking at `Data.Time.Duration` and thought it didn't have much utility, but `Data.DateTime` does.